app/workers/send_user_invitation_worker.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class SendUserInvitationWorker < ApplicationJob
+  queue_as :default
+  include Net
+
+  def perform(invitation_id)
+    invitation = Invitation.find(invitation_id)
+
+    mailer = invitation.account.provider? ? ProviderInvitationMailer : InvitationMailer
+    mailer.invitation(invitation).deliver_now!
+
+    invitation.update!(sent_at: Time.zone.now)
+  rescue OpenSSL::SSL::SSLError, Net::SMTPError, SocketError, ActiveRecord::RecordNotFound => err
+    logger.error(err.message)
```

Comment:
We could use here https://github.com/3scale/porta/blob/master/app/lib/system/error_reporting.rb#L7

Reply:
That notifies to bugsnag, and does not retry.

What we should have is:

- ActiveRecord::RecordNotFound => Do not notify to bugsnag, and do not retry
- OpenSSL::SSL::SSLError, Net::SMTPError, SocketError => Do not notify to bugsnag because it is not a bug in our side, and retry the job.

Reply:
Agreed.

Reply:
In order to retry, the exception has to be risen again, right? And if rescued, the job will be discarded.